src/pages/settings/server/Bans.tsx

```diff
@@ -75,24 +83,33 @@ export const Bans = observer(({ server }: Props) => {
                 <div className={styles.virtual}>
                     <Virtuoso
                         totalCount={data.bans.length}
-                        itemContent={(index) => (
-                            <Inner
-                                key={data.bans[index]._id.user}
-                                server={server}
-                                users={data.users}
-                                ban={data.bans[index]}
-                                removeSelf={() => {
-                                    setData({
-                                        bans: data.bans.filter(
-                                            (y) =>
-                                                y._id.user !==
-                                                data.bans[index]._id.user,
-                                        ),
-                                        users: data.users,
-                                    });
-                                }}
-                            />
-                        )}
+                        itemContent={(index) =>
+                            (!query ||
+                                data.users
+                                    .find(
+                                        (u) =>
+                                            u._id == data.bans[index]._id.user,
+                                    )
+                                    ?.username.toLowerCase()
```

Comment:
This is a pretty big performance penalty which should be memo-ized.

Reply:
Like this?

Reply:
`useMemo`, `useCallback` just prevents the function from being recreated, while `useMemo` keeps the result between renders and also you don't call it every time you render an item

Reply:
TIL